File: ios/App/Sources/Core/Domain/NavCaddy/Round.swift
import Foundation

/// Represents a golf round.
struct Round: Codable, Hashable, Identifiable {
    let id: String
    let startTime: Date
    let courseName: String?

    /// Map of hole number to score
    let scores: [Int: Int]

    init(
        id: String = UUID().uuidString,
        startTime: Date = Date(),
        courseName: String? = nil,
        scores: [Int: Int] = [:]
    ) {
        self.id = id
        self.startTime = startTime
        self.courseName = courseName
        self.scores = scores
    }
}

File: android/app/src/main/java/com/example/app/domain/navcaddy/models/Round.kt
package caddypro.domain.navcaddy.models

import com.google.gson.annotations.SerializedName

/**
 * Domain model representing a golf round.
 *
 * Used for session context and score tracking.
 *
 * Spec reference: navcaddy-engine.md R6
 *
 * @property id Unique identifier for the round
 * @property startTime Unix timestamp when round started
 * @property courseName Name of the course (optional)
 * @property scores Map of hole number to score
 */
data class Round(
    @SerializedName("id")
    val id: String,

    @SerializedName("start_time")
    val startTime: Long,

    @SerializedName("course_name")
    val courseName: String? = null,

    @SerializedName("scores")
    val scores: Map<Int, Int> = emptyMap()
) {
    init {
        scores.keys.forEach { hole ->
            require(hole in 1..18) { "Invalid hole number: $hole" }
        }
    }
}
